src/Illuminate/Validation/Rules/Url.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use Stringable;
+
+class Url implements Stringable
+{
+    protected ?array $protocols = null;
```

Comment:
You might want to type this array:
```suggestion
    /**
     * @var string[]|null $protocols
     */
    protected ?array $protocols = null;
```

Reply:
Cool, I also added docblock for the method. https://github.com/laravel/framework/pull/56405/commits/30c2137bc88dc51c958ebc68885f3f52932af24e

Reply:
Nice! Thanks a lot!